pkg/scheduler/algorithm/priorities/metadata.go

```diff
@@ -123,32 +123,58 @@ func getSelector(pod *v1.Pod, sl corelisters.ServiceLister, cl corelisters.Repli
 		}
 	}
 
-	if rcs, err := cl.GetPodControllers(pod); err == nil {
-		for _, rc := range rcs {
-			labelSet = labels.Merge(labelSet, rc.Spec.Selector)
+	rcPod, rsPod, stsPod := false, false, false
+	for _, ref := range pod.OwnerReferences {
```

Comment:
only pay attention to references with `Controller != nil && *Controller == true`. once you find a reference of that type, you can break the loop (there is at most one reference like that)

Reply:
addressed